lib/cocoapods/downloader/cache.rb

```diff
@@ -156,6 +157,34 @@ def self.valid_lock?(file, filename)
 
       private
 
+      def checksum_for_directory(directory)
```

Comment:
how "expensive" (slow) can this get? should we provide a cache keyed by `directory`?

Reply:
> how "expensive" (slow) can this get?

Well, it didn't have any noticeable impact on test execution time, so it seems fairly cheap. I would assume that the main bottleneck for MD5 is the I/O and not CPU. Considering that we are already doing other I/O operations (like downloading and copying) in connection with this, I think it is fair to assume that the files being checked are already cached in RAM. I suppose, in theory, you could have a problem with pods that are too big to fit in RAM, but pods are generally small and RAM is generally plentiful, so in practice, that should never be a problem.

> should we provide a cache keyed by `directory`?

I'm afraid I don't understand the question. :)